
<launch>
    <param name="use_sim_time" value="true"/>
	<!-- tf -->
    <include file="$(find making_local_map)/launch/tf.launch" />

	<!-- velodyne -->
    <include file="$(find velodyne_height_map)/launch/amsl_velodyne_heightmap.launch" />
    <include file="$(find velodyne_pointcloud)/launch/32e_points.launch" />

	<!-- hokuyo -->
	<node pkg="making_local_map" type="scan2pcl" name="scan2pcl" output="screen">
		<remap from="/scan" to="/front_lasr/scan"/>
		<remap from="/cloud" to="/hokuyo_obstacles"/>
	</node>


	<!-- make each occupancygrid -->
		<!-- velodyne -->
	<node pkg="making_local_map" type="ground_and_obstacle_to_grid" name="velodyne_localmap">
		<remap from="/rm_ground" to="/velodyne_obstacles"/>
		<remap from="/ground" to="/velodyne_clear"/>
		<remap from="/occupancygrid/lidar" to="/occupancygrid/velodyne"/>
	</node>
		<!-- realsense -->
	<node pkg="making_local_map" type="ground_and_obstacle_to_grid" name="realsense_localmap">
		<remap from="/rm_ground" to="/velodyne_obstacles"/>
		<remap from="/ground" to="/camera/depth_registered/points"/>
		<remap from="/occupancygrid/lidar" to="/occupancygrid/realsense"/>
	</node>
		<!-- hokuyo -->
	<node pkg="making_local_map" type="ground_and_obstacle_to_grid" name="hokuyo_localmap">
		<remap from="/rm_ground" to="/hokuyo_obstacles"/>
		<remap from="/ground" to="/velodyne_clear"/>
		<remap from="/occupancygrid/lidar" to="/occupancygrid/hokuyo"/>
	</node>

	<!-- integrate localmap -->
	<node pkg="making_local_map" type="occupancygrid_integrate" name="occupancygrid_integrate">
		<remap from="/occupancygrid/lidar/stored" to="/occupancygrid/velodyne"/>
		<remap from="/occupancygrid/realsense" to="/occupancygrid/realsense"/>
		<remap from="/occupancygrid/hokuyo" to="/occupancygrid/hokuyo"/>
	</node>

	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="/home/amsl/rviz/test.rviz"/>

	<!-- rosbag play -->
	<node pkg="rosbag" type="play" name="play" args="--clock /home/amsl/bagfiles/realsense_2019-05-16-16-41-13.bag -l" />
</launch>
